src/interceptors/ClientRequest/utils/normalizeHttpRequestParams.ts

```diff
@@ -60,7 +60,7 @@ export function normalizeHttpRequestParams(
     debug('created request options', options)
 
     callback = resolveCallback(args)
-  } else if ('method' in args[0]) {
+  } else if (Object.prototype.toString.call(args[0]) === '[object Object]') {
```

Comment:
Does this result in correct typing in this if statement's body? I don't think that TypeScript would like this comparison.

Reply:
Typescript does not complain.

Reply:
I've added a `isRequestOptions` function in PR #63. Do you think that the approach there is better, or do you feel that the check would be too lax, for this use case?

https://github.com/mswjs/node-request-interceptor/pull/63/files#diff-dd33cf0eafb9cac75538d7fd8fb5be0147f2224d1e471400203a8a28ce5947fdR34

Reply:
> Typescript does not complain.

I know, I'm more interested if it can correctly determine the `args[0]` and other types within this closure. Can it?

Reply:
It infers that the type is `RequestOptions`, if I change the `else if` to an `else`, because the first two blocks are matching the other two types. If I keep the third `else if` as an `else if`, but change the condition to something which matches no three of the types, like `else if(typeof args[0] === 'number')`, then it resolves the type of args[0] as `never`. 

If change the condition to something like `Object.prototype.toString.call(args[0]) === 'blah'`, then it still infers `args[0]` as `RequestOptions`. So it seems that it's deciding that it is `RequestOptions` because we are in an `else` and the other two possibilities have been caught by the other blocks, and that it is not using _this_ style of condition to infer the type at all.

In short, it has decided that we have a `RequestOptions` object _unless_ the condition explicitly makes this impossible.

Reply:
I guess since the previous condition was assuming we had an object at that stage, it could be argued that since RequestOptions has no mandatory properties, perhaps the original `else-if` should be swapped for an `else` and the exception should be removed?

Reply:
@sean-hernon so you think it's safe to remove the last `else` clause with an exception, because now this check handles any given object?

Reply:
I would say it's safe to remove, because since we already use the `in` operator on the argument at this point, we are dealing with something object-like, so even if someone passed a function or an array, then it would simply result in lots of `undefined` in ClientRequest and the defaults would be used. Checking that we have an actual plain `Object object` is explicitly checking that we're likely to have something more sane, though.